app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java

```diff
@@ -10,70 +10,78 @@
-            }
-        }
+    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
+        return runCommonTextAction(action);
     }
 
     @Override
-    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
-        return runCommonTextAction(action);
+    protected ActionCallback getActionCallback(@StringRes int keyId) {
+        return new MarkdownTextActionsImpl(keyId);
+    }
+
+    @Override
+    public List<ActionItem> getActiveActionList() {
+        return Arrays.asList(TMA_ACTIONS);
+    }
+
+    @Override
+    protected @StringRes int getFormatActionsKey() {
+        return R.string.pref_key__markdown__action_keys;
     }
 
     //
     //
     //
 
-
-    // Mapping from action (string res) to icon (drawable res)
-    private static final int[][] TMA_ACTIONS = {
-            {R.string.tmaid_common_checkbox_list, R.drawable.ic_check_box_black_24dp, R.string.check_list},
-            {R.string.tmaid_common_unordered_list_char, R.drawable.ic_list_black_24dp, R.string.unordered_list},
-            {R.string.tmaid_markdown_bold, R.drawable.ic_format_bold_black_24dp, R.string.bold},
-            {R.string.tmaid_markdown_italic, R.drawable.ic_format_italic_black_24dp, R.string.italic},
-            {R.string.tmaid_common_delete_lines, CommonTextActions.ACTION_DELETE_LINES_ICON, R.string.delete_lines},
-            {R.string.tmaid_common_open_link_browser, CommonTextActions.ACTION_OPEN_LINK_BROWSER__ICON, R.string.open_link},
-            {R.string.tmaid_common_attach_something, R.drawable.ic_attach_file_black_24dp, R.string.attach},
-            {R.string.tmaid_common_special_key, CommonTextActions.ACTION_SPECIAL_KEY__ICON, R.string.special_key},
-            {R.string.tmaid_common_time, R.drawable.ic_access_time_black_24dp, R.string.date_and_time},
-            {R.string.tmaid_markdown_code_inline, R.drawable.ic_code_black_24dp, R.string.inline_code},
-            {R.string.tmaid_common_ordered_list_number, R.drawable.ic_format_list_numbered_black_24dp, R.string.ordered_list},
-            {R.string.tmaid_markdown_table_insert_columns, R.drawable.ic_view_module_black_24dp, R.string.table},
-            {R.string.tmaid_markdown_quote, R.drawable.ic_format_quote_black_24dp, R.string.quote},
-            {R.string.tmaid_markdown_h1, R.drawable.format_header_1, R.string.heading_1},
-            {R.string.tmaid_markdown_h2, R.drawable.format_header_2, R.string.heading_2},
-            {R.string.tmaid_markdown_h3, R.drawable.format_header_3, R.string.heading_3},
-            {R.string.tmaid_markdown_horizontal_line, R.drawable.ic_more_horiz_black_24dp, R.string.horizontal_line},
-            {R.string.tmaid_markdown_strikeout, R.drawable.ic_format_strikethrough_black_24dp, R.string.strikeout},
-            {R.string.tmaid_common_accordion, R.drawable.ic_arrow_drop_down_black_24dp, R.string.accordion},
+    private static final ActionItem[] TMA_ACTIONS = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5344500)